client/proxy/proxy.go

```diff
@@ -627,3 +633,154 @@ func HandleTcpWorkConnection(ctx context.Context, localInfo *config.LocalSvrConf
 		xl.Debug("join connections closed")
 	}
 }
+
+type SudpProxy struct {
+	*BaseProxy
+
+	cfg *config.SudpProxyConf
+
+	localAddr *net.UDPAddr
+
+	closeCh chan struct{}
+}
+
+func (pxy *SudpProxy) Run() (err error) {
+	pxy.localAddr, err = net.ResolveUDPAddr("udp", fmt.Sprintf("%s:%d", pxy.cfg.LocalIp, pxy.cfg.LocalPort))
+	if err != nil {
+		return
+	}
+
+	pxy.xl.Info("SudpProxy localip: %v, port: %v", pxy.cfg.LocalIp, pxy.cfg.LocalPort)
+	return
+}
+
+func (pxy *SudpProxy) Close() {
+	pxy.mu.Lock()
+	defer pxy.mu.Unlock()
+	select {
+	case <-pxy.closeCh:
+		return
+	default:
+		close(pxy.closeCh)
+	}
+}
+
+func (pxy *SudpProxy) InWorkConn(conn net.Conn, m *msg.StartWorkConn) {
```

Comment:
In this logic，if we have 2 frpc visitor， 1 frpc，1 frps。 Every frpc visitor has its own TCP connection， And if we clean up the previous connection， it may affect each other。
so I take out the workConn from SudpProxy struct。